it/mongodb/src/test/java/org/apache/beam/it/mongodb/DefaultMongoDBResourceManagerTest.java

```diff
@@ -0,0 +1,310 @@
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.mongodb.MongoBulkWriteException;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.MongoIterable;
+import java.io.IOException;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableList;
+import org.bson.Document;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.testcontainers.containers.MongoDBContainer;
+
+/** Unit tests for {@link DefaultMongoDBResourceManager}. */
+@RunWith(JUnit4.class)
+public class DefaultMongoDBResourceManagerTest {
+
+  @Rule public final MockitoRule mockito = MockitoJUnit.rule();
+
+  @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+  private MongoIterable<String> collectionIterable;
+
+  @Mock private MongoClient mongoClient;
+  @Mock private MongoDatabase database;
+  @Mock private MongoCollection<Document> collection;
+  @Mock private MongoCursor<String> collectionNames;
+  @Mock private MongoDBContainer container;
+
+  private static final String TEST_ID = "test-id";
+  private static final String COLLECTION_NAME = "collection-name";
+  private static final String STATIC_DATABASE_NAME = "database";
+  private static final String HOST = "localhost";
+  private static final int MONGO_DB_PORT = 27017;
+  private static final int MAPPED_PORT = 10000;
+
+  private DefaultMongoDBResourceManager testManager;
+
+  @Before
+  public void setUp() throws IOException, InterruptedException {
+    when(container.getHost()).thenReturn(HOST);
+    when(container.getMappedPort(MONGO_DB_PORT)).thenReturn(MAPPED_PORT);
+
+    testManager =
+        new DefaultMongoDBResourceManager(
+            mongoClient, container, DefaultMongoDBResourceManager.builder(TEST_ID));
+  }
+
+  @Test
+  public void testCreateResourceManagerBuilderReturnsDefaultMongoDBResourceManager()
+      throws IOException {
+    assertThat(
+            DefaultMongoDBResourceManager.builder(TEST_ID)
+                .useStaticContainer()
+                .setHost(HOST)
+                .setPort(MONGO_DB_PORT)
+                .build())
+        .isInstanceOf(DefaultMongoDBResourceManager.class);
+  }
+
+  @Test
+  public void testGetUriShouldReturnCorrectValue() {
+    assertThat(testManager.getUri()).matches("mongodb://" + HOST + ":" + MAPPED_PORT);
+  }
+
+  @Test
+  public void testGetDatabaseNameShouldReturnCorrectValue() {
+    assertThat(testManager.getDatabaseName()).matches(TEST_ID + "-\\d{8}-\\d{6}-\\d{6}");
```

Comment:
All the setup for the test is outside of the test case. You should inline it so that what is being tested is clear. Currently it basically just tests the setup.

Reply:
Refactored the test class to make it more clear. Thanks!